ui/src/app/views/pipeline/show/audit/pipeline.audit.component.ts

```diff
@@ -0,0 +1,288 @@
+    @Input() pipeline: Pipeline;
+
+    audits: Array<PipelineAudit>;
+
+    currentCompare: Array<{ title, before, after, type }>;
+
+    @ViewChild('auditModal')
+    public auditModalTmpl: ModalTemplate<boolean, boolean, void>;
+    modal: ActiveModal<boolean, boolean, void>;
+    modalConfig: TemplateModalConfig<boolean, boolean, void>;
+
+    codeMirrorConfig: any;
+
+    constructor(private _auditService: PipelineAuditService, private _modalService: SuiModalService) {
+        super();
+        this.codeMirrorConfig = {
+            matchBrackets: true,
+            autoCloseBrackets: true,
+            mode: 'application/json',
+            lineWrapping: true,
+            autoRefresh: true,
+            readOnly: 'nocursor'
+        };
+    }
+
+    getData(): any[] {
+        return this.audits;
+    }
+
+    ngOnInit(): void {
+        this._auditService.getAudit(this.project, this.pipeline).first().subscribe(as => {
+            this.audits = as;
+        });
+    }
+
+    compareIndex(i): void {
+        let indexToCompate = (this.currentPage - 1) * this.nbElementsByPage + i;
+
+        let pipFrom = cloneDeep(this.audits[indexToCompate].pipeline);
+        let pipTo: Pipeline;
+        if (indexToCompate === 0) {
+            pipTo = cloneDeep(this.pipeline);
+        } else {
+            pipTo = cloneDeep(this.audits[indexToCompate - 1].pipeline);
+        }
+
+        pipFrom = this.cleanPipeline(pipFrom);
+        pipTo = this.cleanPipeline(pipTo);
+
+        this.currentCompare = new Array<{ title, before, after, type }>();
+        compare(pipFrom, pipTo).forEach(c => {
+            let diff = {
+                before: '',
+                after: '',
+                title: '',
+                type: 'string'
+            };
+
+            let path = c.path;
+            let pathSplitted = path.split('/').filter(p => p !== '');
+
+            switch (this.audits[indexToCompate].action) {
+                case 'addStage':
+                    diff = this.getAddStageDiff(pathSplitted, pipTo);
+                    break;
+                case 'updateStage':
+                    diff = this.getUpdateStageDiff(pathSplitted, pipTo, pipFrom);
+                    break;
+                case 'deleteStage':
+                    diff = this.getDeleteStageDiff(pathSplitted, pipFrom);
+                    break;
+                case 'addJob':
+                    diff = this.getAddJobDiff(pathSplitted, pipTo);
+                    break;
+                case 'updateJob':
+                    diff = this.getUpdateJobDiff(path, pathSplitted, pipTo, pipFrom);
+                    break;
+                case 'deleteJob':
+                    diff = this.getDeleteJobDiff(pathSplitted, pipFrom);
+                    break;
+            }
+
+            if (diff) {
+                this.currentCompare.push(diff);
+            }
+        });
+    }
+
+    getAddJobDiff(path: Array<string>, pipTo: Pipeline): { title, before, after, type } {
```

Comment:
maybe better to create a model class for diff model instead of always write this `{ title, before, after, type }`